app/services/persister_task_service.rb

```diff
@@ -4,12 +4,18 @@ def initialize(options)
 
     validate_options
     @upload_task = Task.find(@options[:category])
+    @source = Source.find_by(:id => @upload_task.source_id)
   end
 
   def process
     return self unless source_enabled?
 
     @task = FullRefreshPersisterTask.create!(opts)
+    @upload_task.update!(:child_task_id => @task.id)
```

Comment:
@hsong-rh Before we create the Persister Task we should check if the @upload_task has errors. There is a race condition here where the ingress comes in before the on premise client can update the task. The ingress message comes via Kafka and the update of the Upload task happens using REST API. So the only safe thing we can do is to check if the UploadTask doesn't have any errors and not wait to see if the upload task has completed.